### PR TITLE
[1LP][RFR] Add Providers with SSL

### PR DESCRIPTION
**Pull request by pavelzag**
This adds SSL verification test to Add Providers and also fixes refresh timeout for one of the tests.
The test currently fails on PRT due to cfme_data.yml discrepancy but passes on our Jenkins

Purpose or Intent
=================
{{pytest: cfme/tests/containers/test_manageiq_ansible_providers.py -v --use-provider cm-env2 }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 623060b562e738359f9a62fefb438d756b2c4fe2:

`utils/ansible.py`:
- :bangbang: Line 170:5: **E303** *too many blank lines (2)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b09076b48bf60dab9112ec929f551947b68c0e0c:

`cfme/tests/containers/test_manageiq_ansible_providers.py`:
- :bangbang: Line 126:1: **E303** *too many blank lines (3)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/containers/test_manageiq_ansible_providers.py [hunk 1]
```diff
@@ -87,27 +112,38 @@ def test_manageiq_ansible_update_provider_incorrect_user(ansible_providers, prov
...
-    run_ansible('remove_provider')
+    script_name = 'remove_provider'
+    setup_ansible_script(provider, script_type='providers', script=script_name)
+    run_ansible(script_name)
     navigate_to(ContainersProvider, 'All', use_resetter=True)
-    assert sel.is_displayed_text('No Records Found.')
+    text_to_assert = 'No Records Found.'
+    soft_assert(
+        wait_for(
+            lambda: assertion(text_to_assert),
```
Comment (quarckster) [inline cfme/tests/containers/test_manageiq_ansible_providers.py]: **REQUIRED**
Widgetastic here should be used here.
```python
view = navigate_to(ContainersProvider, 'All', use_resetter=True)
soft_assert(
    wait_for(
        lambda: not view.table.is_displayed,
        num_sec=120, delay=10,
        fail_func=view.browser.refresh,
        message='Provider was not deleted successfully',
        silent_failure=True)
)
```

File: cfme/tests/containers/test_manageiq_ansible_providers.py [hunk 2]
```diff
@@ -31,6 +32,27 @@ def ansible_providers():
     remove_tmp_files()
 
 
+def assertion(text):
+    if sel.is_displayed_text(text):
+        return True
+    return False
```
Comment (quarckster) [inline cfme/tests/containers/test_manageiq_ansible_providers.py]: **REQUIRED**
This function is not needed.

File: cfme/tests/containers/test_manageiq_ansible_providers.py [hunk 3]
```diff
@@ -69,10 +93,11 @@ def test_manageiq_ansible_add_provider_same_name(ansible_providers, provider):
...
 
         """
-    setup_ansible_script(provider, script_type='providers', script='add_provider')
-    run_ansible('add_provider')
-    run_ansible('add_provider')
-    assert get_yml_value('add_provider', 'name') in provider.summary.properties.name.value
+    script_name = 'add_provider'
+    setup_ansible_script(provider, script_type='providers', script=script_name)
+    run_ansible(script_name)
+    run_ansible(script_name)
```
Comment (quarckster) [inline cfme/tests/containers/test_manageiq_ansible_providers.py]: Why is it required twice?

**Comment by dajoRH** [comment]
Lint report for commit 2f6027762b1d2f3a1a869312c1a8c2629a3a5924:

`cfme/tests/containers/test_manageiq_ansible_providers.py`:
- :bangbang: Line 77:12: **E127** *continuation line over-indented for visual indent*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/containers/test_manageiq_ansible_providers.py (hunk 3 above)
Comment (pavelzag) [reply cfme/tests/containers/test_manageiq_ansible_providers.py]: This test checks adding adding Containers Provider with the same name

File: cfme/tests/containers/test_manageiq_ansible_providers.py [hunk 4]
```diff
@@ -87,27 +107,38 @@ def test_manageiq_ansible_update_provider_incorrect_user(ansible_providers, prov
...
-    assert sel.is_displayed_text('No Records Found.')
+    script_name = 'remove_provider'
+    setup_ansible_script(provider, script_type='providers', script=script_name)
+    run_ansible(script_name)
+    view = navigate_to(ContainersProvider, 'All', use_resetter=True)
+    soft_assert(
+        wait_for(
+            lambda: not get_yml_value(script_name, 'name') in view.entities.entity_names,
+            num_sec=180, delay=10,
+            fail_func=view.browser.refresh(),
```
Comment (quarckster) [inline cfme/tests/containers/test_manageiq_ansible_providers.py]: **REQUIRED**
`fail_func` must be a callable: `view.browser.refresh`

**Comment by quarckster** [comment]
All tests failed in PRT.

**Comment by pavelzag** [comment]
Yes, as stated in the title, they fail due to misconfiguration on cfme_data.yml.
The tests pass on our local Jenkins though

**Pull request merged**